docs/designs/Compilation.md

```diff
@@ -0,0 +1,20 @@
+# Page compilation
+
+The compilation system for ASP.NET Framework was MSBuild based and not something that we needed to bring-forward as-is as it was mostly an implementation detail. Instead, we are using Roslyn and creating compilations based on that.
+
+## IWebFormsCompiler
+
+The `IWebFormsCompiler` interface is used to access the heavy lifting of taking a collection of input files and compiling that aspx/ascx/etc into their appropriate types. It takes an `ICompilationStrategy` that is used to configure what the compilation should look like. The main implementations of this are for in-memory (i.e. runtime) compilation compared to static compilation (i.e. aspnet_compiler).
+
+## PageCompilationOptions
+
+This type is used to configure various aspects of the general compilation system. This includes:
+
+- *Parsers* - The implementations of various `DependencyParser` for a given extension. This type has been augmented from the ASP.NET Framework version to include the ability to generate a code parser and then a code generator from that as a starting point.
```

Comment:
See if you can add about determining compilation dependency and how we sort them before invoking parsers/generators.